### ISSUE TITLE
Zeus on iOS expects HTTPS for LND REST; does not support plain HTTP even over Tor

### ISSUE THREAD

**Issue by philipmw** [ZeusLN/zeus#2525]
### Describe the bug

(**edit:** see comments; this has evolved.)

I run LND over Tor. It's at http://sv7ld5ajw6qtipmcttpzhhihhjpjfqs662cin36ylvh3sa6abv2fg3qd.onion.

When I try to set up Zeus with it ("LND over REST" mode), Zeus is never able to connect. It spins for ~30 seconds and says, "Error: An SSL error has occurred and a secure connection to the server cannot be made."

I've tried both built-in Tor support and Tor via Orbot app. Same result with both.

How I know the service is working properly:

1. On the same phone, I can open the onion URL in my browser. It gets as far as a 404 with a JSON response.
2. On my laptop, I am able to configure Alby browser extension with the same onion address, and Lightning functionality is working.

### Reproduce

1. Node interface: LND (REST)
2. Host: `sv7ld5ajw6qtipmcttpzhhihhjpjfqs662cin36ylvh3sa6abv2fg3qd.onion`
3. Macaroon: ...
4. REST port: 80
5. Use Tor: off
6. Certificate Verification: off

Another way:

1. Node interface: LND (REST)
2. Host: `sv7ld5ajw6qtipmcttpzhhihhjpjfqs662cin36ylvh3sa6abv2fg3qd.onion`
3. Macaroon: ...
4. REST port: 80
5. Use Tor: on

### ZEUS version

v0.9.1

### Node interface

LND (REST)

### Network

Tor

### Device

iPhone 12 Pro

### Device operating system

iOS 18.1

**Comment by philipmw** [comment]
Today I updated to 0.9.2. The problem remains.

**Comment by kaloudis** [comment]
> Today I updated to 0.9.2. The problem remains.

have you tried restarting LND or the Tor process on your node?

**Comment by philipmw** [comment]
Yes, both. Happy to try specific things. But also -- my node is publicly (over Tor) reachable. I posted its address in the original report. Want to try it yourself?

**Comment by kaloudis** [comment]
I think the SSL error would only throw if you had Certificate verification enable when using with Orbot. You want this unchecked as Tor services typically don't have https enabled, plus yours seems to have a http prefix.

If issues persist you should consider an alternative remote connection method like Lightning Node Connect or Tailscale. We have some guides available here: https://docs.zeusln.app/category/remote-connections/

**Comment by philipmw** [comment]
I agree that it is suspicious that the error mentions SSL. I have cert verification **off**, and the REST endpoint does not use TLS. So Zeus shouldn't be attempting TLS at all.

**Comment by philipmw** [comment]
I confirmed that the issue is that Zeus is assuming/expecting TLS.

As an experiment, I changed my REST endpoint to use TLS, and I reconfigured the Tor onion service to listen on port 443 instead of 80. Zeus was able to connect, using the built-in Tor support!

So, the bug is that Zeus does not support plain HTTP for the LND REST mode, and expects HTTPS even over Tor. I will update the issue title.

**Comment by kaloudis** [comment]
Following up on this: plain HTTP for LND REST over Tor is now supported and has been verified working on a device.

To use it, enter the scheme explicitly in the host field, e.g. `http://sv7...fg3qd.onion`, with your REST port and built-in Tor enabled. Bare hosts (no scheme) still default to `https://`, which we're keeping as the secure default; that's why the original config produced an SSL error.

What changed since this was filed:
- The Tor stack was replaced with react-native-nitro-tor (#3971), whose HTTP client carries plain `http://` requests over Tor natively.
- HTTPS onion services with self-signed certs now also work out of the box: TLS certificate validation is bypassed for `.onion` HTTPS endpoints over Tor, since the onion address already authenticates the peer (strict TLS is kept for clearnet-over-Tor). So the TLS-on-443 workaround described above no longer requires any cert configuration.
- The new cleartext HTTP warning (#4363) deliberately exempts `.onion` hosts, as the Tor layer provides transport encryption.

Closing as resolved. Thanks for the detailed report and for narrowing down the root cause.
